updates to the dependency table

woodbe opened the pull request:
This is to close #347

Taking a good look at this, it is a lot of text. I'm rewriting it to look more like the Network Device table where the rationale statements are simpler (things like "SFR included" or other simple statements).

I'm adding DSC-specific dependencies in italics in the lists so they are clearly different than the CC:2022 ones.

One question here is that the catalog excludes FCS_CKM.3 from the dependencies. There is FCS_CKM_EXT.3 which is similar and probably sufficient in general. I'm adding this in for some but not all, though it would seem to mean that that not having it included in any FCS_COP.1 requires an explanation.

Looking at the catalog, the following dependencies seem wrong (in the catalog, the table or both):

FCS_COP.1/KeyedHash - it would seem that FCS_COP.1/Hash is a dependency, since SHA is required for HMAC-SHA to work. Also, it lists FCS_CKM.5 as a dependency, but that seems backwards. Further, several of the other ones don't make a lot of sense.

FCS_COP.1/CMAC - shouldn't FCS_COP.1/SKC be a dependency here? Somehow though FCS_COP.1/SigGen is, but this doesn't have anything to do with signature generation. I used the same as from FCS_COP.1/AEAD, but I'm not sure some of those line up right.

FCS_COP.1 in CC:2022 has a dependency on FCS_CKM.3. I have added this to all the SFRs and point it to the FCS_CKM_EXT.3 with a note.

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,391 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
 
 |FCS_CKM.1 
-|[FCS_CKM.2, FCS_CKM.5, FCS_CKM_EXT.7 or FCS_COP.1] 
+|[_FCS_CKM.1/AKG_, _FCS_CKM.1/SKG_, FCS_CKM.2, FCS_CKM.5, FCS_CKM_EXT.7, _FCS_CKM_EXT.8_ or FCS_COP.1] 
```
FCS_CKM_EXT.7 is not in CC:2022, is it? I also don't see it being required by us in FCS_CKM.1

woodbe replied:
In the crypto catalog they have it listed, since clearly it can't be there from cc:2022 as you say. Similarly FCS_CKM.1 can't be a dependency on itself since it would be circular in the line below (but that is how we wrote the requirement).

Basically the list of dependencies is a combination of the CC:2022 and the crypto catalog (plus, where applicable, what we may have added)

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,391 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.2 - included
+
+FCS_CKM.3 is satisfied by FAU_STG_EXT.1 which controls access to keys - included
```
Is FCS_CKM.5 (selection-based) missing?

woodbe replied:
added (oops)

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,391 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FDP_ITC.2 is satisfied by FDP_ITC_EXT.2 related to a mechanism by which key data can be imported into the TSF - included
+
+FCS_CKM.1 - included
```
Aren't FDP_ITC.1, FDP_ITC.2, and FCS_CKM.1 selection-based too?

jvdsn replied:
Unless selection-based here refers to the SFR in the DSC as opposed to a selection in FCS_CKM.2

woodbe replied:
selection-based is that the SFR itself is selection-based, not what is a selection within the SFR. This is basically what the Network Device cPP looked like (though they said "selection-based SFR" and I left off the SFR).

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,391 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.2 - included
+
+FCS_CKM.3 is satisfied by FAU_STG_EXT.1 which controls access to keys - included
```
I guess we can't say FCS_CKM.3 is satisfied by FCS_CKM_EXT.3 because FCS_CKM_EXT.3 is selection-based?

woodbe replied:
Sort of. If you look at what FCS_CKM.3 is talking about, it is access to the keys. In our cPP, FCS_STG_EXT.1 is what actually specifies how access control to the keys is handled, and one of the options from there is FCS_CKM_EXT.3, but also it could be hardware-controlled (as opposed to needing to be encrypted). So to me it seems like that was the actual better linkage in terms of how we meet the expectations for the CC:2022 dependencies.

Note that in almost all the crypto catalog they have removed FCS_CKM.3 as a dependency. I was adding it back because I didn't want to not list things that were in the CC:2022.

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,391 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+FCS_COP.1/AEAD - (selection-based) included
+
+|FCS_CKM.6 
+|[FDP_ITC.1, FDP_ITC.2, FCS_CKM.1, FCS_CKM_EXT.3, FCS_CKM.5, FCS_CKM_EXT.7 or FCS_CKM_EXT.8] 
```
From what I read in CC:2022, FCS_CKM_EXT.3, FCS_CKM.5, FCS_CKM_EXT.7 or FCS_CKM_EXT.8 are not dependencies of FCS_CKM.6. Should they be italicized?

woodbe replied:
@slpotte I added these because they are in the catalog. There is a note (to Frank) to add text as to why they are added. I expect they are added because at least some of them relate to how the keys are protected in the first place, so they provide some sort of protection and justification as to why it is OK that keys are deleted in a specific manner.

If this isn't available, I don't have a problem, for now, just taking out those and leaving only the raw CC:2022 dependencies.

woodbe replied:
For now I am going to remove these dependencies

jvdsn reviewed (changes requested):
I plan to review all changes once those few comments are answered so I can understand the process (i.e. what's an actual issue and what I'm misreading).

woodbe commented:
Remove italics for the cPP-added dependencies

For the crypto catalog:
- maybe the changes from CC:2022 need to have a sentence about why they are added
- maybe they should be called out with italics?
- maybe anything with changed dependencies should reference the catalog (except that the catalog isn't published yet)?

slpotte reviewed (commented):
For FCS_COP.1/KeyedHash, we added FCS_COP.1/Hash to the list of dependencies, and removed references to hash algorithms from the list of standards.

For FCS_COP.1/CMAC - I missed that.  I will add the dependency as suggested, and remove all the symmetric algorithm standards from the list of standards.

For FCS_COP.1, the dependency FCS_CKM.3 has been removed according to the errata (see link below) - so there no need to replace it.

https://www.commoncriteriaportal.org/files/ccfiles/CCMB-2024-002-v1_1-Errata_Interpretation_CC_CEM_2022.pdf

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_COP.1/Hash
+
+FCS_CKM.3
```
Errata 1 (CC2022-P2-R1-0007) removed the FCS_CKM.3 dependency

woodbe replied:
done

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_COP.1/Hash - included
+
+FCS_CKM.3 is satisfied by FAU_STG_EXT.1 which controls access to keys - included
```
Errata 1 (CC2022-P2-R1-0007) removed the FCS_CKM.3 dependency

woodbe replied:
done

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+|FCS_COP.1/KeyEncap
+|[FDP_ITC.1, FDP_ITC.2, FCS_CKM.1, FCS_CKM.5, FCS_CKM_EXT.7, or FCS_CKM_EXT.8]
+
+FCS_CKM.3
```
Errata 1 (CC2022-P2-R1-0007) removed the FCS_CKM.3 dependency

woodbe replied:
done

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.3
 
 FCS_CKM.6 
```
The latest version of the Crypto Catalog that I have also includes FCS_OTV_EXT.1

woodbe replied:
updated

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM_EXT.8 - (selection-based SFR) included
+
+FCS_CKM.3 is satisfied by FAU_STG_EXT.1 which controls access to keys - included
```
Errata 1 (CC2022-P2-R1-0007) removed the FCS_CKM.3 dependency

woodbe replied:
done

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
 |FCS_COP.1/KeyWrap
 |[FDP_ITC.1, FDP_ITC.2, FCS_CKM.1, FCS_CKM.5, FCS_CKM_EXT.7, or FCS_CKM_EXT.8]
 
+FCS_CKM.3
```
Errata 1 (CC2022-P2-R1-0007) removed the FCS_CKM.3 dependency

woodbe replied:
done

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM_EXT.8 - (selection-based SFR) included
+
+FCS_CKM.3 is satisfied by FAU_STG_EXT.1 which controls access to keys - included
```
Errata 1 (CC2022-P2-R1-0007) removed the FCS_CKM.3 dependency

I won't be making any more comments on this for the other FCS_COP.1 iterations

woodbe replied:
done

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
 
 |FCS_COP.1/SigGen 
-|[FDP_ITC.1, FDP_ITC.2, FCS_CKM.1, or FCS_CKM.5] 
+|[FDP_ITC.1, FDP_ITC.2, FCS_CKM.1/AKG, or FCS_CKM.5] 
```
The latest version of the Crypto Catalog that I have does not include FCS_CKM.5. Or is this one taken from CC:2022?

woodbe replied:
it is in the latest version of the catalog that I have and it is in CC:2022 for FCS_COP.1.

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+FCS_CKM.6 - included
 
 |FCS_COP.1/SigVer 
 |[FDP_ITC.1, FDP_ITC.2, FCS_CKM.1, or FCS_CKM.5] 
```
The latest version of the Crypto Catalog that I have does not include FCS_CKM.1 or FCS_CKM.5. Or are these two taken from CC:2022?

woodbe replied:
They are from CC:2022. There is a note in the catalog of "we need to provide a justification about deleting them". While I don't dispute it being unnecessary, for simplicity at this point we have it there, they are selections, and so it is low as far as whether it is a concern to me at this point.

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
 |FDP_FRS_EXT.1 
 |FDP_FRS_EXT.2 
-|FDP_FRS_EXT.2 is an optional SFR in the PP. It does not need to be required because the PP's definition of FDP_FRS_EXT.1 allows the ST author to specify that no factory reset is possible, in which case the behavior described by FDP_FRS_EXT.2 is not triggered.
+|FDP_FRS_EXT.2 - (optional-based SFR) included
```
I feel like "optional-based" isn't really the right wording?

woodbe replied:
changed

jvdsn replied:
it should actually be selection-based

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.6 - included
+
+|FCS_COP.1/KeyEncap
```
This is a selection-based SFR, so it doesn't belong in this table.

woodbe replied:
moved

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.6 - included
 
 |FCS_COP.1/KeyWrap
```
This is a selection-based SFR, so it doesn't belong in this table.

woodbe replied:
moved

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4229,34 +4407,42 @@ FMT_SMR.1
-|FCS_RBG.1 is required by this PP. FCS_RBG.5 is required when combining multiple entropy sources.
+|FCS_RBG.1 - included
+
+FCS_RBG.5 - (selection-based SFR) included
```
Is optional, not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4229,34 +4407,42 @@ FMT_SMR.1
+
+FCS_RBG.3 - (selection-based SFR) included
+
+FCS_RBG.4 - (selection-based SFR) included
```
All of these (except FCS_RBG.1) are optional, not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
+
+FCS_CKM.5 - included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.5 - (selection-based SFR) included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.5 - (selection-based SFR) included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.5 - (selection-based SFR) included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.5 - (selection-based SFR) included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+|FCS_RBG.2 - (selection-based SFR) included
+
+FCS_RBG.3 - (selection-based SFR) included
```
Both FCS_RBG.2 and FCS_RBG.3 are optional, not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
+
+FCS_COP.1/KeyWrap - (selection-based SFR) included
+
+FCS_COP.1/SKC - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
-FDP_SDC.2 
-|All of FCS_CKM.1, FCS_COP.1, and FDP_SDC.2 are required by the PP.
+FCS_CKM.6
+|FCS_CKM.1 - included
```
Should be FCS_CKM.2.

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
-|FCS_CKM.1/KeyedHash and FCS_CKM.6 are required by the PP. FDP_ITC_EXT.1 and FDP_ITC_EXT.2 are also required by the PP and equivalent to FDP_ITC.1 in this case because they all relate to a mechanism by which key data can be imported into the TSF.
+|FCS_CKM.2 - included
+
+FCS_COP.1 - included
```
Should be removed

woodbe replied:
removed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
+
+FCS_COP.1 - included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
+
+FCS_CKM.1/KeyedHash - included
```
Should be FCS_COP.1/KeyedHash

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
+
+FCS_CKM.5 - (selection-based SFR) included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
+
+FCS_CKM.6
+
+FCS_OTV_EXT.1
```
The Crypto Catalog revision that I have includes FCS_COP.1/SigGen here instead of FCS_OTV_EXT.1 but I'm 95% sure that's an error. SigGen wouldn't make sense as a dependency.

woodbe replied:
Remove FCS_OTV_EXT.1

woodbe replied:
removed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
+
+FCS_CKM.5 - (selection-based SFR) included
+
+FCS_CKM_EXT.7 - (selection-based SFR) included
```
is not selection-based

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4279,97 +4469,211 @@ FPT_ROT_EXT.1
-FCS_COP.1 
-|FPT_MFW_EXT.1 and FCS_COP.1 are required by the PP.
+FCS_COP.1
+|FPT_MFW_EXT.1- included
```
Add space before hyphen.

woodbe replied:
fixed

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4265,7 +4451,11 @@ FCS_RBG.5
+
+FPT_PRO_EXT.1- included
+
+FPT_ROT_EXT.1- included
```
Please add spaces before all hyphens

woodbe replied:
fixed

woodbe commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
 
 |FCS_CKM.1 
-|[FCS_CKM.2, FCS_CKM.5, FCS_CKM_EXT.7 or FCS_COP.1] 
+|[FCS_CKM.1/AKG, FCS_CKM.1/SKG, FCS_CKM.2, FCS_CKM.5, FCS_CKM_EXT.7, FCS_CKM_EXT.8 or FCS_COP.1] 
```
I removed all the italics

woodbe replied:
@woodbe Need to remove mention of the italics

woodbe replied:
done

woodbe commented on cPP_DSC.adoc:
```diff
@@ -3107,15 +3107,14 @@ There are no auditable events foreseen.
 [horizontal]
 Hierarchical to:: No other components.
 
-Dependencies:: [FDP_ITC.1 Import of user data without security attributes, or +
```
In the latest one it has been removed.

woodbe commented:
I just realized that I had removed the FCS_CKM.3 from all these when I did the original errata review, and then went back through and added them all back in here (DOH!). Oh well.

woodbe commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
-|[FCS_CKM.2, FCS_CKM.5, FCS_CKM_EXT.7 or FCS_COP.1] 
+|[FCS_CKM.1/AKG, FCS_CKM.1/SKG, FCS_CKM.2, FCS_CKM.5, FCS_CKM_EXT.7, FCS_CKM_EXT.8 or FCS_COP.1] 
+
+FCS_CKM.3
```
done

woodbe commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
 
-|FCS_CKM.1 is required by the PP. FDP_ITC_EXT.1 and FDP_ITC_EXT.2 are also required by the PP and equivalent to FDP_ITC.1 in this case because they all relate to a mechanism by which key data can be imported into the TSF.
+|FCS_CKM.6 
+|[FDP_ITC.1, FDP_ITC.2 or FCS_CKM.1] 
```
done

woodbe commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+|FCS_COP.1/KeyedHash 
+|[FDP_ITC.1, FDP_ITC.2, FCS_CKM.1, FCS_CKM.5, FCS_CKM_EXT.7 or FCS_CKM_EXT.8] 
+
+FCS_COP.1/Hash
```
updated

jvdsn commented on cPP_DSC.adoc:
```diff
@@ -4019,203 +4015,385 @@ The dependencies between SFRs implemented by the TOE are addressed as follows.
+
+FCS_CKM.2 - included
+
+FCS_CKM.3 is satisfied by FAU_STG_EXT.1 which controls access to keys - included
```
I still see "FCS_CKM.3 is satisfied by FAU_STG_EXT.1 which controls access to keys - included"

woodbe replied:
removed (thought I got them all)

woodbe commented:
updated the last two comments.
